internal/api/proxy.go

```diff
@@ -0,0 +1,139 @@
+package api
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/net/proxy"
+)
+
+func applyProxy(transport *http.Transport, proxyEndpointURL *url.URL, proxyEndpointPath string) (applied bool) {
+	if proxyEndpointURL == nil && proxyEndpointPath == "" {
+		return false
+	}
+
+	handshakeTLS := func(conn net.Conn, addr string) (net.Conn, error) {
+		// Extract the hostname (without the port) for TLS SNI
+		host, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			return nil, err
+		}
+		// Be sure to clone TLS-specific config settings from the transport,
+		// like InsecureSkipVerify.
+		tlsConn := tls.Client(conn, &tls.Config{
+			ServerName:         host,
+			InsecureSkipVerify: transport.TLSClientConfig.InsecureSkipVerify,
+		})
+		if err := tlsConn.Handshake(); err != nil {
+			return nil, err
+		}
```

Comment:
We should use the `HandshakeContext` variant here

Reply:
Yep, good catch!